canopen/node/local.py

```diff
@@ -127,3 +130,39 @@ def _find_object(self, index, subindex):
                 raise SdoAbortedError(0x06090011)
             obj = obj[subindex]
         return obj
+
+    def _nmt_state_changed(self, old_state, new_state):
+        if new_state == "OPERATIONAL":
+            for i, pdo in self.tpdo.map.items():
+                if pdo.enabled:
+                    try:
+                        pdo.start()
+                        logger.info(f"Successfully started TPDO {i}")
+                    except ValueError:
+                        logger.warning(f"Failed to start TPDO {i} due to missing period")
+                    except Exception as e:
+                        logger.error(f"Unknown error starting TPDO {i}: {str(e)}")
+                else:
+                    logger.info(f"TPDO {i} not enabled")
+        elif old_state == "OPERATIONAL":
+            self.tpdo.stop()
+
+    def _tpdo_configuration_write(self, index, subindex, od, data):
+        if 0x1800 <= index <= 0x19FF:
+            # Only allowed to edit pdo configuration in pre-op
+            if self.nmt.state != "PRE-OPERATIONAL":
```

Comment:
That's not true in general, but we could make it a requirement for this library's implementation.  Please indicate clearly if it should be the latter.

Reply:
I almost had it, depending on what level we want to support (variable vs dynamic) it can be changed in pre-op or operational. [https://www.can-cia.org/can-knowledge/pdo-protocol-1](https://www.can-cia.org/can-knowledge/pdo-protocol-1#:~:text=in%20CANopen%C2%A0FD.-,PDO%20mapping,-CANopen%20FD%20distinguishes). I expanded it to allow while in operational. This still leaves out the PDO mapping procedure...

Reply:
The only restriction I found in the standard is that subindex 03 "shall not be changed while the PDO exists (bit 31 ofsub-index 01h is set to 0b)."  So as long as the library's implementation does support dynamic reconfiguration (which I think it does?) we have no reason to restrict its usage IMHO.

The mentioned restriction as well as the PDO mapping procedure should be handled in the PDO code, but that's definitely not a blocker for this PR.